lib/Driver/UnixToolChains.cpp

```diff
@@ -165,7 +168,10 @@ toolchains::GenericUnix::constructInvocation(const DynamicLinkJobAction &job,
 
   // Select the linker to use.
   std::string Linker;
-  if (const Arg *A = context.Args.getLastArg(options::OPT_use_ld)) {
+  if (context.OI.LTOVariant != OutputInfo::LTOKind::None) {
+    // Force to use lld for LTO
+    Linker = "lld";
```

Comment:
Why should we force the use of lld here? Will this run on Darwin? On Darwin we definitely want to use ld64.

Reply:
This will run on Unix-like platform (Linux, Android or etc not including Darwin). And on these platforms, we don't support gold LTO or something else LTO except for lld. So I forced the use of lld here.

For example, gold LTO needs libLLVMGold.so but we don't have it in swift toolchain and it's licensed under GPL, so it's a little complex problem.

At this time, I only support lld on Unix-like platforms.